### DISCUSSION TITLE
New User Experience

### DISCUSSION THREAD

**Discussion by nattaylor** [ActivityWatch/activitywatch#798] [General]
ActivityWatcher is wonderful.  I have failed at many manual time based trackers, and ActivityWatcher's use of window titles feels magical.

On MacOS 12.6 (21G115) running Version:  v0.12.0 I have also been plagued by the problem `Exception thrown while trying to get active window`. I am having _some_ luck with the work arounds proposed, like launching via Terminal but still it is inconsistent in ways that I haven't exactly pinned down -- perhaps issues with the permissions surviving restarts or just Macs being temperamental.

In the image below you can see by looking at the day-over-day barchart that `aw-watcher-window` works for short periods, then starts throwing excepts again.  ActivityWatcher itself is running this whole time.  I blame this on MacOS but I haven't been able to track down what happens.  I also use Espanso which can get temporarily locked out by MacOS's "secure input"  feature, so perhaps something similar is happening here.

<img width="749" alt="image" src="https://user-images.githubusercontent.com/2789593/191729027-9ae427f9-1a29-4897-8808-e708c860cd28.png">

I am also having the issue with my hostname switching from Nats-MBP.home to Nats-MPB, but from another ticket it seems that has to do my how my own system is configured.

Still, despite that, I find ActivityWatch a delight to use as it is revealing interesting time spent data to me.

Thank you to the contributors!

**Comment by BelKed** [comment]
Around a month ago there was released a new version of ActivityWatch ([`0.12.1`](https://github.com/ActivityWatch/activitywatch/releases/tag/v0.12.1)) which solved problems with launching the application itself (use of Terminal is not needed now) and even issues with tracking active window

Feel free to upgrade to the latest version and try it :)
